test/spec/controllers/courtSearchInfo.ctrl.spec.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+describe('CourtSearchInfoCtrl', function() {
+  var CourtSearchInfoCtrl;
+  var sample_court;
+
+  sample_court = {
+    id: 9,
+    latitutde: 38.7947506,
+    longitude: -90.26261696,
+    municipality: "Black Jack",
+    address: "12500 Old Jamestown Road",
+    city: "St. Louis",
+    state: "MO",
+    zip_code: 63033
+  };
+
+  beforeEach(function() {
+    module('yourStlCourts');
+
+    inject(function($controller,$state, $window, $q, $httpBackend){
+
+      CourtSearchInfoCtrl = $controller('CourtSearchInfoCtrl',{
+        $state: $state,
+        $window: $window,
+        court:sample_court
+      });
+
+      $httpBackend.whenGET(/courts/).respond(200, '');
+    });
+  });
+
+  xit('goes back to home page if court is null',inject(function($state){
+    spyOn($state,'go');
+    sample_court = null;
+    expect($state.go).toHaveBeenCalledWith('home');
+  }));
+
+  it('prints if printCourtInfo is called',inject(function($window){
+    spyOn($window,'print');
+    CourtSearchInfoCtrl.printCourtInfo();
+    expect($window.print).toHaveBeenCalled();
+  }));
+
+  it('sets court on initialization',inject(function($rootScope){
+    $rootScope.$apply();
+    expect(CourtSearchInfoCtrl.courtInfo).toEqual(sample_court);
+  }));
+
+  it('calls and gets correct directions',(function(){
```

Comment:
I suppose my question was this:  In line 34 of the controller, the code already has: ctrl.courtDirectionLink  = ctrl.getCourtDirectionLink(ctrl.courtInfo);
Why can't I expect that to have been called?  When I test with the existing code on lines 58 & 60 of the test code, the courtDirectionLink has been correctly set to the directionString so it must have been called.